.github/workflows/deploy-gcp-tests.yml

```diff
@@ -196,24 +196,42 @@ jobs:
-      # or at least that the local state version is greater than the available cached state version from main.
+      # Find a cached state disk for this job, matching all of:
+      # - disk kind (disk_prefix)
+      # - disk target height (disk_suffix)
+      # - state version (from the source code)
       #
-      # Aftwards, get the disk name to be used on further steps
-      - name: Validate local state version with cached state
+      # If there are multiple disks:
+      # - prefer images generated from the `main` branch, then any other branch
+      # - prefer newer images to older images
+      #
+      # Passes the disk name to subsequent steps using an environmental variable.
+      - name: Find cached state disk
         id: get-disk-name
         run: |
           LOCAL_STATE_VERSION=$(grep -oE "DATABASE_FORMAT_VERSION: .* [0-9]+" "$GITHUB_WORKSPACE/zebra-state/src/constants.rs" | grep -oE "[0-9]+" | tail -n1)
           echo "LOCAL_STATE_VERSION: $LOCAL_STATE_VERSION"
 
-          ZEBRA_STATE_DISK=$(gcloud compute images list --filter="name~${{ inputs.disk_prefix }} AND name~-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
-          echo "Disk: $ZEBRA_STATE_DISK"
-          echo "Description: $(gcloud compute images describe $ZEBRA_STATE_DISK --format='value(DESCRIPTION)')"
+          # Try to find an image generated from the main branch
+          # name~${{ inputs.disk_prefix }}-main- AND
+          ZEBRA_STATE_DISK=$(gcloud compute images list --filter="name~-v${LOCAL_STATE_VERSION}- AND name~-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
+          echo "main Disk: $ZEBRA_STATE_DISK"
+
+          if [[ -z "$ZEBRA_STATE_DISK" ]]; then
+              # Try to find an image generated from any other branch
+              # name~${{ inputs.disk_prefix }}- AND
+              ZEBRA_STATE_DISK=$(gcloud compute images list --filter="name~-v${LOCAL_STATE_VERSION}- AND name~-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
+              echo "Disk: $ZEBRA_STATE_DISK"
+          fi
 
-          GCP_STATE_VERSION=$(echo "$ZEBRA_STATE_DISK" | grep -oE "v[0-9]+" | grep -oE "[0-9]+")
-          echo "GCP_STATE_VERSION: $GCP_STATE_VERSION"
+          if [[ -z "$ZEBRA_STATE_DISK" ]]; then
+              echo "No cached state disk available"
+              echo "Expected ${{ inputs.disk_prefix }} v$LOCAL_STATE_VERSION ${{ inputs.disk_suffix }}"
+              echo "Cached state test jobs must depend on the cached state rebuild job"
+              exit 1
+          fi
 
-          if [[ "$LOCAL_STATE_VERSION" -lt "$GCP_STATE_VERSION" ]]; then echo "Local version is lower than cached version" && exit 1; fi
+          echo "Description: $(gcloud compute images describe $ZEBRA_STATE_DISK --format='value(DESCRIPTION)')"
 
           echo "ZEBRA_CACHED_DISK_NAME=$ZEBRA_STATE_DISK" >> $GITHUB_ENV
```

Comment:
```suggestion
          echo "CACHED_DISK_NAME=$STATE_DISK" >> $GITHUB_ENV
```

Reply:
Fixed in 30a12c1c8 to just use `CACHED_DISK_NAME` everywhere.
(Using two names makes it more likely we'll choose the wrong one in subsequent steps.)